sdl_viewer/src/terrain_drawer/read_write.rs

```diff
@@ -0,0 +1,56 @@
+use crate::graphic::tiled_texture_loader::{TilePos, TiledTextureLoader};
+use cgmath::Vector3;
+use image::{LumaA, Rgba};
+use point_viewer::math::Isometry3;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io::{Error, ErrorKind};
+
+#[derive(Serialize, Deserialize)]
+pub struct Metadata {
+    pub tile_size: u32,
+    pub world_from_terrain: Isometry3<f64>,
+    pub origin: Vector3<f64>,
+    pub resolution_m: f64,
+    pub tile_positions: Vec<TilePos>,
+}
+
+pub type TextureLoaders = (TiledTextureLoader<LumaA<f32>>, TiledTextureLoader<Rgba<u8>>);
+
+#[allow(dead_code)]
+impl Metadata {
+    // Custom serialization of some values
+    pub fn from_dir<P: AsRef<std::path::Path>>(dir: P) -> Result<Self, Error> {
+        let meta_path = dir.as_ref().join("meta.json");
+        let reader = File::open(meta_path)?;
+        let meta = serde_json::from_reader(reader).map_err(|e| {
```

Comment:
nit: return here instead of using an intermediate variable `meta`?

Reply:
Don't you think the parenthesizing would hurt readibility a bit if we wrap it in a Ok()?

Reply:
`serde_json::from_reader` returns a `Result` so I don't think we have to wrap in in `Ok`, or do we?

Reply:
Oh yeah